### ISSUE TITLE
`complete` external command wrapped in parenthesis

### ISSUE THREAD

**Issue by jaredramirez** [nushell/nushell#9738]
### Describe the bug

I have an external command that takes lots of arguments and I want to run that in a Nu script and take action if it fails using `complete`. To make the external command's arguments easier to manage, I broke it out over multiple lines. To do so, I have to wrap the command in `()`. This causes the `complete` function to error with:

```
Error:   × Complete only works with external streams
    ╭─[...]
 22 │     }
 23 │       | complete
    ·         ────┬───
    ·             ╰── complete only works on external streams
 24 │   )
    ╰────
```

### How to reproduce

Attempt to run:
```
 (
    do {
      (^ls
        -al
      )
    }
      | complete
  )
```

If I squash the external command into 1 line and remove the `()`, it works:
```
(
    do {
      ^ls -al
    }
      | complete
  )
```

My real-world external command is using Nu to script postgres database action and looks like:
```
  let result = (
    do {
      PGOPTIONS='--client-min-messages=warning' ^psql $conn_str -X -q -b -v ON_ERROR_STOP=1 --pset pager=off
        -f my_sql_file
        -v option_a="$OPTION_A"
        -v option_b="$OPTION_B"
        -v const_c="my_constant"
        ... and so on for 15-20 more lines
    }
      | complete
  )
```

### Expected behavior

I expect to be able to `complete` an external command wrapped in parenthesis

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.80.0                                                            |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | macos-x86_64                                                      |
| build_target       | x86_64-apple-darwin                                               |
| rust_version       | rustc 1.69.0 (84c898d65 2023-04-16) (built from a source tarball) |
| cargo_version      | cargo 1.69.0                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| features           | default, zip                                                      |
| installed_plugins  |                                                                   |

**Comment by amtoine** [comment]
that's a good catch @jaredramirez, can definitely reproduce with
```nushell
> do { (git st) } | complete
fatal: not a git repository (or any of the parent directories): .git
Error:   × Complete only works with external streams
   ╭─[entry #1:1:1]
 1 │ do { (git st) } | complete
   ·                   ────┬───
   ·                       ╰── complete only works on external streams
   ╰────
```
outside of a Git repo :+1:

**Comment by futile** [comment]
Can confirm this still happens in version 0.85.0

**Comment by amtoine** [comment]
also on `0.85.1`

**Comment by KAAtheWiseGit** [comment]
The issue is that parens can't return raw input, so they convert the command output to a string:

```Nushell
~/fork/nushell> ^ls | describe 
raw input
~/fork/nushell> (^ls) | describe 
string
```

So the fix would involve changing how subexpression works, and not the semantics of `complete`.

**Comment by KAAtheWiseGit** [comment]
@amtoine, right!

And I did a quick check after Your comment, those are indeed different.  `into string` adds/preserves an extra newline.

**Comment by tingerrr** [comment]
Running `do { ^external } | complete` for a non-existing `external` also raises the stream error, instead of raising command not found.

**Comment by KAAtheWiseGit** [comment]
@tingerrr it will throw the correct error in 0.88.2 or 0.89.  Got fixed at #11313

**Comment by IanManske** [comment]
This should be fixed in version 0.92. Parentheses are for grouping only now and no longer collect external command output. Let us know if anything still looks off.
